kroxylicious-filters/kroxylicious-record-encryption/src/test/java/io/kroxylicious/filter/encryption/encrypt/InBandEncryptionManagerTest.java

```diff
@@ -68,6 +68,29 @@ void testMultipleThreadsCooperateToMinimiseDekCreations() {
         assertThat(cache.invalidationCount()).isEqualTo(numEncryptionOperations - 1);
     }
 
+    @Test
+    void shouldGrowBuffer() {
+        // Given
+        InMemoryKms kms = getInMemoryKms();
+        ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+        final DekManager<UUID, InMemoryEdek> dekManager = new DekManager<>(new AsyncKms<>(kms, executor), 10000);
+        EncryptionDekCache<UUID, InMemoryEdek> cache = new EncryptionDekCache<>(dekManager, executor, EncryptionDekCache.NO_MAX_CACHE_SIZE, Duration.ofHours(1),
+                Duration.ofHours(1));
+        var encryptionManager = createEncryptionManager(dekManager, cache, executor);
+        var kekId = kms.generateKey();
+        var value = new byte[3 * 1024 * 1024];
```

Comment:
Give we're testing how the buffer growth interacts with dek allocation, I think we should be commenting something about the magic numbers in this test.

Reply:
have renamed the var and added a constant for the initial case size so it's defined in terms of that